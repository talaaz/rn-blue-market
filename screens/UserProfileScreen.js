import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { Button, Snackbar } from "react-native-paper";

const UserProfileScreen = (props) => {
  const [visible, setVisible] = React.useState(false);

  const onToggleSnackBar = () => setVisible(!visible);

  const onDismissSnackBar = () => setVisible(false);

  return (
    <View style={styles.container}>
      <Text>UserProfileScreen</Text>
      <Button onPress={onToggleSnackBar}>{visible ? "Hide" : "Show"}</Button>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});

export default UserProfileScreen;
