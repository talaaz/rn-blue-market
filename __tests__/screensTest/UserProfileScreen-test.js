import React from "react";
import renderer from "react-test-renderer";
import UserProfileScreen from "../../screens/UserProfileScreen";

jest.useFakeTimers();
describe("<UserProfileScreen />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<UserProfileScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
