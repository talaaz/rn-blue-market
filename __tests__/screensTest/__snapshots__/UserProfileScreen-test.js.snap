// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfileScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f5f5f5",
      "flex": 1,
      "justifyContent": "space-evenly",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "System",
          "fontWeight": "400",
          "textAlign": "left",
        },
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "System",
            "fontWeight": "500",
            "textAlign": "left",
            "writingDirection": "ltr",
          },
          Array [
            Object {
              "fontSize": 20,
              "letterSpacing": 0.15,
              "lineHeight": 30,
              "marginVertical": 2,
            },
            undefined,
          ],
        ],
      ]
    }
  >
    Not Logged In
  </Text>
</View>
`;
